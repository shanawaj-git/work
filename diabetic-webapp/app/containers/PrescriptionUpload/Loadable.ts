/**
 * Asynchronously load the component from Upload Prescription Screen
 */
import loadable from 'utils/loadable';
export default loadable(() => import('./index'));
