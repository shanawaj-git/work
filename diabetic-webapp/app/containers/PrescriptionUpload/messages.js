/**
 * Prescription Upload screen messages
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  prescriptionHeaderPrefixLabel: {
    id: 'prescription.upload.header.label.part1',
    defaultMessage: 'Scan & Upload',
  },
  prescriptionHeaderSuffixLabel: {
    id: 'prescription.upload.header.label.part2',
    defaultMessage: 'your prescription',
  },
  prescriptionDescriptionLabel: {
    id: 'prescription.upload.description.label',
    defaultMessage:
      'Grab your prescription, upload it and proceed to setting up your delivery.',
  },
  prescriptionUploadButton: {
    id: 'prescription.upload.button.label',
    defaultMessage: 'Upload Picture',
  },
});
