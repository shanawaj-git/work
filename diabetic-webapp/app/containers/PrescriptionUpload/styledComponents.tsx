import styled from 'styled-components';

export const UploadPrescriptionImage = styled.div`
  direction: rtl;
`;
export const UploadPrescription = styled.div`
  display: flex;
  flex-direction: column;
  background: white;
  min-height: 100vh;
  min-width: 100%;
  position: fixed;
`;

export const UploadPrescriptionHeader = styled.div`
  display: block;
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 32px;
`;

export const UploadPrescriptionDescription = styled.div`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  color: black;
  margin: 20px 0px 20px 0px;
`;

export const PrescriptionContent = styled.div`
  margin: 20px 20px 20px 20px;
`;
export const PrescriptionUploadButton = styled.button`
  color: white;
  position: absolute;
  bottom: 100px;
  height: 48px;
  width: 90%;
  /* BG/Primary */
  background: #00664f;
  border-radius: 12px;
`;

export const PrescriptionInput = styled.input`
  display: none;
`;
