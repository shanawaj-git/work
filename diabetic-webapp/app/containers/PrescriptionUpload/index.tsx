import React, { useRef, useState } from 'react';
import PropTypes, { any, InferProps } from 'prop-types';
import { injectIntl } from 'react-intl';
import { Typography, TypographyType } from '@albathanext/design-system';
import {
  PrescriptionUploadButton,
  PrescriptionContent,
  UploadPrescription,
  UploadPrescriptionDescription,
  UploadPrescriptionHeader,
  UploadPrescriptionImage,
  PrescriptionInput,
} from './styledComponents';
import messages from './messages';
const uploadPrescriptionImage = require('images/educationImage2.svg');
const PRESCRIPTION_REVIEW_SCREEN = 'review-prescription';
function PrescriptionUpload({ intl, history }: PrescriptionUploadTypes) {
  const inputRef = useRef(null);

  const handleClick = () => {
    inputRef.current.click();
  };
  function handleFileChange(event) {
    const fileObj = event.target.files && event.target.files[0];
    if (!fileObj) {
      return;
    }
    const prescriptionImage = fileObj;
    history.push({
      pathname: PRESCRIPTION_REVIEW_SCREEN,
      state: { prescriptionImage },
    });
  }

  return (
    <UploadPrescription>
      <UploadPrescriptionImage>
        <img src={uploadPrescriptionImage} alt="upload-prescription-img" />
      </UploadPrescriptionImage>
      <PrescriptionContent>
        <UploadPrescriptionHeader>
          <Typography typographyType={TypographyType.TITLE_1}>
            {intl.formatMessage(messages.prescriptionHeaderPrefixLabel)}
          </Typography>
          <Typography typographyType={TypographyType.LARGE_TITLE}>
            {intl.formatMessage(messages.prescriptionHeaderSuffixLabel)}
          </Typography>
        </UploadPrescriptionHeader>
        <UploadPrescriptionDescription>
          <Typography typographyType={TypographyType.SUB_HEAD}>
            {intl.formatMessage(messages.prescriptionDescriptionLabel)}
          </Typography>
        </UploadPrescriptionDescription>
        <PrescriptionUploadButton
          onClick={handleClick}
          data-testid="prescription-button"
        >
          <PrescriptionInput
            data-testid="prescription-input"
            ref={inputRef}
            type="file"
            accept="image/*"
            onChange={handleFileChange}
          />
          {intl.formatMessage(messages.prescriptionUploadButton)}
        </PrescriptionUploadButton>
      </PrescriptionContent>
    </UploadPrescription>
  );
}

const PrescriptionUploadPropTypes = {
  intl: any,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
};

type PrescriptionUploadTypes = InferProps<typeof PrescriptionUploadPropTypes>;

PrescriptionUpload.propTypes = PrescriptionUploadPropTypes;

export default injectIntl(PrescriptionUpload);
