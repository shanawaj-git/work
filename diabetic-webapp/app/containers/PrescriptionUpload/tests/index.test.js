import { fireEvent, render } from 'react-testing-library';
import { act } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import React from 'react';
import PrescriptionUpload from '../index';
jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/prescription-upload',
    search: '',
    hash: '',
    state: null,
    key: '',
  }),
}));

describe('<PrescriptionUpload />', () => {
  it('should render and match the snapshot', () => {
    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <PrescriptionUpload {...props} />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should select and upload the prescription when upload button is clicked', async () => {
    global.URL.createObjectURL = jest.fn();
    const mockHistory = {
      push: jest.fn(),
    };
    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    act(() => {
      const { getByTestId } = render(
        <IntlProvider locale="en">
          <PrescriptionUpload history={mockHistory} {...props} />
        </IntlProvider>,
      );

      const fakeFile = new File(['hello'], 'hello.png', { type: 'image/png' });
      const inputFile = getByTestId(/prescription-input/i);
      const uploadButton = getByTestId(/prescription-button/i);

      fireEvent.click(uploadButton);
      fireEvent.change(inputFile, {
        target: { files: [fakeFile] },
      });

      expect(mockHistory).toBeDefined();
      expect(inputFile).toBeDefined();
    });
  });

  it('should not proceed to the next screen when no image file is selected', async () => {
    const mockHistory = {
      push: jest.fn(),
    };
    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    act(() => {
      const { getByTestId } = render(
        <IntlProvider locale="en">
          <PrescriptionUpload history={mockHistory} {...props} />
        </IntlProvider>,
      );

      const inputFile = getByTestId(/prescription-input/i);
      const uploadButton = getByTestId(/prescription-button/i);

      fireEvent.click(uploadButton);
      fireEvent.change(inputFile, {
        target: { files: [undefined] },
      });

      expect(mockHistory).toBeDefined();
      expect(mockHistory.push).not.toBeCalled();
    });
  });
});
