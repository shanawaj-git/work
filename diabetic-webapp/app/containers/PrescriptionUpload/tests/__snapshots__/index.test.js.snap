// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrescriptionUpload /> should render and match the snapshot 1`] = `
<div
  class="sc-bwzfXH efIfpC"
>
  <div
    class="sc-bdVaJa kgLnZe"
  >
    <img
      alt="upload-prescription-img"
      src="IMAGE_MOCK"
    />
  </div>
  <div
    class="sc-ifAKCX hahzy"
  >
    <div
      class="sc-htpNat jnDXAh"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1jfg05l-MuiTypography-root"
      >
        Scan & Upload
      </h2>
      <h1
        class="MuiTypography-root MuiTypography-h1 css-qu1isr-MuiTypography-root"
      >
        your prescription
      </h1>
    </div>
    <div
      class="sc-bxivhb gXuoul"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1reifv3-MuiTypography-root"
      >
        Grab your prescription, upload it and proceed to setting up your delivery.
      </p>
    </div>
    <button
      class="sc-EHOje bIniWJ"
      data-testid="prescription-button"
    >
      <input
        accept="image/*"
        class="sc-bZQynM fqeBUw"
        data-testid="prescription-input"
        type="file"
      />
      Upload Picture
    </button>
  </div>
</div>
`;
